src/python/pants/option/parser.py

```diff
@@ -576,9 +578,15 @@ def expand(val_or_str):
                 if val_or_str.startswith("@@"):  # Support a literal @ for fromfile values via @@.
                     return val_or_str[1:]
                 else:
-                    fromfile = val_or_str[1:]
+                    if val_or_str.startswith("@?"):  # Support an optional fromfile value via @?.
+                        fromfile, optional = val_or_str[2:], True
+                    else:
+                        fromfile, optional = val_or_str[1:], False
+                    fromfile_path = Path(get_buildroot(), fromfile)
                     try:
-                        contents = Path(get_buildroot(), fromfile).read_text()
+                        if optional and not fromfile_path.exists():
+                            return None
```

Comment:
I believe it may be useful to log an info/warn message for user, but I don't see any usage in this file already so I'm unsure if it's idiomatic to do at this level for pants?

Reply:
I do think logging a info message here is beneficial. In larger teams, we can't assume that all users are aware of all optional configs (perhaps not the same team managing the configuration as the one later running some pants command).

The motivation for this feature was to not hinder all use of pants merely for a missing unrelated config file; but it _is_ needed for a particular goal, so I'd argue it is not always optional, but as pants is a load all config or nothing kind of thing, there's not difference for if a certain subsystem is being used or not, thus all config must be valid at all times. I think this is the real issue, actually, just being tapered over with optional config values like this.

Reply:
You have to opt in to this via adding the `?` to the fromfile syntax. So I think if you do that you've explicitly said "it's OK for this file not to exist" and so logging that fact will just get in the way.

Reply:
See below for my question on what the intention behind this feature is, because that relates to my opinion on whether we should log or not...